frontend/purposeful-frontend/components/Registration.js

```diff
@@ -14,11 +14,93 @@ import {
 } from "@chakra-ui/react";
 import { ViewIcon, ViewOffIcon } from "@chakra-ui/icons";
 import { Field, Form, Formik } from "formik";
+import fetchWrapper, { getAuthorities } from "../utils/fetch_wrapper";
+import notification from "../utils/notification";
+
+async function handleRegistrationForm(values, actions, GrantedAuth) {
+  const payload = {
+    email: values.email,
+    password: values.password,
+    firstname: values.firstname,
+    lastname: values.lastname,
+  };
+  let response = null;
+
+  if (
+    GrantedAuth.length === 0 &&
+    !window.location.pathname.includes("moderator")
+  ) {
+    // User is not authenticated and is registering as a regular user
+    response = await fetchWrapper(
+      "/api/appuser/regular",
+      null,
+      "POST",
+      payload
+    );
+  } else if (
+    GrantedAuth.includes("Owner") &&
+    window.location.pathname.includes("moderator")
```

Comment:
Otherwise, they are an owner on the moderator page, creating a moderator

Reply:
Correct